src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -2388,9 +2428,15 @@ private static void addMetadataToResponse(HttpServletResponse response,
             response.addHeader(HttpHeaders.ETAG, maybeQuoteETag(
                     BaseEncoding.base16().lowerCase().encode(contentMd5Bytes)));
         }
-        Date expires = contentMetadata.getExpires();
-        if (expires != null) {
-            response.addDateHeader(HttpHeaders.EXPIRES, expires.getTime());
+        String overrideExpires = request.getParameter(
+                ResponseHeaderOverrides.RESPONSE_HEADER_EXPIRES);
+        if (overrideExpires != null) {
+            response.addHeader(HttpHeaders.EXPIRES, overrideExpires);
+        } else {
+            Date expires = contentMetadata.getExpires();
+            if (expires != null) {
+                response.addDateHeader(HttpHeaders.EXPIRES, expires.getTime());
+            }
```

Comment:
This one is not straightforward. To avoid unnecessary evaluations you'll have to use the `first.or(supplier)` pattern, plus the fact that you want `expires.getTime()` not just `expires`. I don't see this becoming more readable than it currently is.